ext/node/ops/sqlite/database.rs

```diff
@@ -9,42 +9,145 @@ use std::ffi::CString;
 use std::ptr::null;
 use std::rc::Rc;
 
+use deno_core::cppgc;
 use deno_core::op2;
-use deno_core::serde_v8;
 use deno_core::v8;
+use deno_core::v8_static_strings;
 use deno_core::GarbageCollected;
 use deno_core::OpState;
 use deno_permissions::PermissionsContainer;
 use rusqlite::ffi as libsqlite3_sys;
+use rusqlite::ffi::SQLITE_DBCONFIG_DQS_DDL;
+use rusqlite::ffi::SQLITE_DBCONFIG_DQS_DML;
 use rusqlite::limits::Limit;
-use serde::Deserialize;
 
 use super::session::SessionOptions;
+use super::statement::check_error_code2;
+use super::validators;
 use super::Session;
 use super::SqliteError;
 use super::StatementSync;
-use crate::ops::sqlite::SqliteResultExt;
+
 const SQLITE_DBCONFIG_ENABLE_LOAD_EXTENSION: i32 = 1005;
 const SQLITE_DBCONFIG_ENABLE_ATTACH_WRITE: i32 = 1021;
 
-#[derive(Deserialize)]
-#[serde(rename_all = "camelCase")]
 struct DatabaseSyncOptions {
-  #[serde(default = "true_fn")]
   open: bool,
-  #[serde(default = "true_fn")]
   enable_foreign_key_constraints: bool,
   read_only: bool,
-  #[serde(default = "false_fn")]
   allow_extension: bool,
+  enable_double_quoted_string_literals: bool,
 }
 
-fn true_fn() -> bool {
-  true
-}
+impl DatabaseSyncOptions {
+  fn from_value(
+    scope: &mut v8::HandleScope,
+    value: v8::Local<v8::Value>,
+  ) -> Result<Self, validators::Error> {
+    use validators::Error;
+
+    if value.is_undefined() {
+      return Ok(Self::default());
+    }
+
+    let Ok(obj) = v8::Local::<v8::Object>::try_from(value) else {
+      return Err(Error::InvalidArgType(
+        "The \"options\" argument must be an object.",
+      ));
+    };
+
+    let mut options = Self::default();
+
+    v8_static_strings! {
+      OPEN_STRING = "open",
+      ENABLE_FOREIGN_KEY_CONSTRAINTS_STRING = "enableForeignKeyConstraints",
+      READ_ONLY_STRING = "readOnly",
+      ALLOW_EXTENSION_STRING = "allowExtension",
+      ENABLE_DOUBLE_QUOTED_STRING_LITERALS_STRING = "enableDoubleQuotedStringLiterals",
+    }
+
+    let open_string = OPEN_STRING.v8_string(scope).unwrap();
```

Comment:
Unrelated to this PR, but I think the API should be `OPEN_STRING.from_scope(scope)` - the unwrap shouldn't be necessary here